xbmc/guilib/GUIControlFactory.cpp

```diff
@@ -492,20 +492,20 @@ bool CGUIControlFactory::GetAnimations(TiXmlNode *control, const CRect &rect, in
 
 bool CGUIControlFactory::GetActions(const TiXmlNode* pRootNode, const char* strTag, CGUIAction& action)
 {
-  action.m_actions.clear();
+  action.Reset();
   const TiXmlElement* pElement = pRootNode->FirstChildElement(strTag);
   while (pElement)
   {
     if (pElement->FirstChild())
     {
-      CGUIAction::cond_action_pair pair;
+      CGUIAction::CondActionPair pair;
       pair.condition = XMLUtils::GetAttribute(pElement, "condition");
       pair.action = pElement->FirstChild()->Value();
```

Comment:
Add a constructor to `CGUIAction::ConditionPair`, taking the condition + action ?